File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/VB/WPFListBox-VB/WPFComponents/Converters/StringToLowerCaseConverter.vb
Imports System.Globalization
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data

<ValueConversion(GetType(String), GetType(String))> _
 Public Class StringToLowerCaseConverter
  Implements IValueConverter

  Public Function Convert(ByVal value As Object, _
                          ByVal targetType As Type, _
                          ByVal parameter As Object, _
                          ByVal culture As CultureInfo) As Object _
                  Implements System.Windows.Data.IValueConverter.Convert
    Return System.Convert.ToString(value).ToLower()
  End Function

  Public Function ConvertBack(ByVal value As Object, _
                              ByVal targetType As Type, _
                              ByVal parameter As Object, _
                              ByVal culture As CultureInfo) As Object _
                  Implements System.Windows.Data.IValueConverter.ConvertBack
    Return System.Convert.ToString(value)
  End Function
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/Converters/StringToLowerCaseConverter.cs
using System;
using System.ComponentModel;
using System.Globalization;
using System.Windows.Data;

namespace WPFComponents
{
  [ValueConversion(typeof(string), typeof(string))]
  public class StringToLowerCaseConverter : IValueConverter
  {
    public object Convert(object value, Type targetType,
                                    object parameter, CultureInfo culture)
    {
      return System.Convert.ToString(value).ToLower();
    }

    public object ConvertBack(object value, Type targetType,
                                                object parameter, CultureInfo culture)
    {
      return System.Convert.ToString(value);
    }
  }
}
